zipline/pipeline/factors/factor.py

```diff
@@ -833,6 +834,90 @@ def linear_regression(self, target, regression_length, mask=NotSpecified):
             mask=mask,
         )
 
+    @float64_only
+    def winsorize(self,
+                  limits,
+                  inclusive=(True, True),
+                  mask=NotSpecified,
+                  groupby=NotSpecified):
+        """
+        Construct a Factor returns a winsorized row for results. Winsorizing
+        clips the input values to fixed percentiles. The (limits[0])th lowest
+        values are set to the value at the (limits[0])th percentile. The values
+        above the (limits[1])th percentiles are set to the value at the
+        (limits[1])th percentile. This is useful when limiting the impact of
+        extreme values.
+
+        If ``mask`` is supplied, ignore values where ``mask`` returns False
+        when computing row means and standard deviations, and output NaN
+        anywhere the mask is False.
+
+        If ``groupby`` is supplied, compute by partitioning each row based on
+        the values produced by ``groupby``, winsorizing the partitioned arrays,
+        and stitching the sub-results back together.
+
+        Parameters
+        ----------
+        limits : None, tuple of float, optional
```

Comment:
One concrete note: I don't think there's any case where it's useful for us to accept `None` for the boundary parameter.